File: BOJ/Greedy/10162/10162.cpp
// 10162 전자레인지
// https://www.acmicpc.net/problem/10162
#include <iostream>
#include <vector>

using namespace std;

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int T;
    vector<int> answer;
    int btn[3] = {300, 60, 10};

    cin >> T;

    for (int i = 0; i < 3; ++i)
    {
        answer.push_back(T / btn[i]);
        T = T % btn[i];
    }

    if (T)
    {
        cout << -1 << '\n';
    } else
    {
        for (int& a: answer)
        {
            cout << a << ' ';
        }
    }

    return 0;
}

File: BOJ/Greedy/10162/10162.py
# 10162 전자레인지
# https://www.acmicpc.net/problem/10162
t = int(input())

a,b,c = 0, 0, 0

while t != 0:
    if t >= 300:
        t -= 300
        a += 1
    elif t >= 60:
        t -= 60
        b += 1
    elif t >= 10:
        t -= 10
        c += 1
    elif t%10 != 0:
        break

if t != 0:
    print(-1)
else:
    print(a,b,c)